### PR TITLE
8253641: Missing newline in the printout of certain JFR events

### PR DESCRIPTION
**Pull request by mbien**
Trivial fix which adds a missing new line to PrettyWriter::printClassLoader if clazz == null.

PrettyWriter is used for both the jfr CLI tool and also for the impl of the public JFR API behind jdk.jfr.Event::toString().

how to reproduce using JFR CLI:

To generate a JFR dump for testing, simply start HelloWorld.java
```
public class HelloWorld {
    public static void main(String[] args) {System.out.println("hello there");}
}
```
with
`jdk-15+36/bin/java -XX:StartFlightRecording=filename=dump.jfr,dumponexit=true HelloWorld.java`

current result for events where the classLoader field equals null (note: newline missing):

```
jdk-15+36/bin/jfr print --events jdk.ModuleRequire dump.jfr
...
jdk.ModuleRequire {
  startTime = 16:12:43.524
  source = {
    name = "jdk.nio.mapmode"
    version = "15"
    location = "jrt:/jdk.nio.mapmode"
    classLoader = null  }
  requiredModule = {
    name = "java.base"
    version = "15"
    location = "jrt:/java.base"
    classLoader = null  }
}
...
```

after the patch, using same jfr dump and displaying the same event:

```
jdk16/build/linux-x86_64-server-release/jdk/bin/jfr print --events jdk.ModuleRequire dump.jfr
...
jdk.ModuleRequire {
  startTime = 16:12:43.524
  source = {
    name = "jdk.nio.mapmode"
    version = "15"
    location = "jrt:/jdk.nio.mapmode"
    classLoader = null
  }
  requiredModule = {
    name = "java.base"
    version = "15"
    location = "jrt:/java.base"
    classLoader = null
  }
}
...
```

sidenote:

There would be an opportunity to merge the two sequential if-else blocks[1] in printClassLoader over the clazz variable after this patch, but this would obfuscate the actual bug fix, so I decided to omit it. Let me know if I should add that.
[1] https://github.com/mbien/jdk/blob/67750e2b599b3f2a6b814d7bceada785ca4051f9/src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java#L440-L448

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253641](https://bugs.openjdk.java.net/browse/JDK-8253641): Missing newline in the printout of certain JFR events

### Reviewers
 * [Markus Grönlund](https://openjdk.java.net/census#mgronlun) (@mgronlun - **Reviewer**)

### Contributors
 * Michael Bien `<mbien42@gmail.com>`

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/248/head:pull/248`
`$ git checkout pull/248`

### CODE REVIEW

**Comment by mbien** [comment]
Tiny PR looking for reviewers - now even with an issue assigned to it.

**Comment by mgronlun** [comment]
Hi Michael, I think you can go ahead and merge the blocks.

Thanks
Markus

**Comment by mgronlun** [comment]
You will also need to align the name of this PR with the synopsis of the JIRA issue, the bots complain about:

Integration blocker

 ⚠️ Title mismatch between PR and JBS for issue JDK-8253641

**Comment by mbien** [comment]
@mgronlun Thank you for the review, I made some changes:

 - changed the original fix so that the postFix is printed in every case after comparing it with similar methods (please review again)
 - added commit which merges the two consecutive ifs over same variable
 - added JDK- prefix to this issue but the bot hasn't noticed yet

File: src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java [hunk 1]
```diff
@@ -437,13 +437,16 @@ void printObject(RecordedObject object, long arraySize) {
...
         RecordedClass clazz = cl.getType();
-        print(clazz == null ? "null" : clazz.getName());
         if (clazz != null) {
+            print(clazz.getName());
             print(" (");
             print("id = ");
             print(String.valueOf(cl.getId()));
-            println(")");
+            print(")");
+        }else{
```
Comment (mgronlun) [inline src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java:446]: please add space after }

File: src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java [hunk 2]
```diff
@@ -437,13 +437,16 @@ void printObject(RecordedObject object, long arraySize) {
...
+            print("null");
         }
+        println(postFix);
```
Comment (mgronlun) [inline src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java:449]: Categorical postFix now possible by changing 445 from println(")"); to print(")");?

Comment (mbien) [reply src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java:449]: yes exactly. I think this is how this method was supposed to work. I compared it with similar methods like printClass().

If there is a postFix, its always printed, and it adds a new line in this case. The new line was conditional before (which caused the bug in the first place).

File: src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java (hunk 1 above)
Comment (mbien) [reply src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java:446]: oh sorry about that, I forgot it this time - old habit. Will fix it.

Comment (mbien) [reply src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java]: fixed

**Review by mgronlun — APPROVED** [review]
Thanks @mbien for fixing this.

**Comment by mbien** [comment]
@mgronlun no problem. Thank you for the review!

**Comment by mgronlun** [comment]
You can issue a comment "/contributor mbien" and then another comment "/integrate".  I will then sponsor your change for integration.

**Comment by mbien** [comment]
(sorry for the spam, the bot was stubborn)

**Comment by mgronlun** [comment]
@mbien no problem, we are all new to the process here. Sorry for providing some vague instructions.

**Pull request closed without merging**